x-pack/plugins/apm/server/lib/transaction_groups/fetcher.ts

```diff
@@ -72,9 +72,18 @@ export function transactionGroupsFetcher(setup: Setup, bodyQuery: StringMap) {
     }
   };
 
-  if (esFilterQuery) {
-    params.body.query.bool.filter.push(esFilterQuery);
-  }
+  params.body.query = {
+    ...params.body.query,
+    bool: {
+      ...params.body.query.bool,
+      filter: [
+        ...(params.body.query.bool && params.body.query.bool.filter
+          ? params.body.query.bool.filter
+          : []),
+        ...uiFiltersES
+      ]
+    }
+  };
```

Comment:
Probably simpler to add `uiFiltersES` in the two callers (I think you were going to do that originally, right?)
https://github.com/elastic/kibana/blob/aded18bfb835b1c9132fd5f47f0d2dbd5ee76516/x-pack/plugins/apm/server/lib/transactions/get_top_transactions/index.ts#L51

https://github.com/elastic/kibana/blob/aded18bfb835b1c9132fd5f47f0d2dbd5ee76516/x-pack/plugins/apm/server/lib/traces/get_top_traces.ts#L33

Reply:
Not sure i totally grok what's going on with these abstractions. It might be best to address some of this as tech debt?

Reply:
I don't blame you :D I agree that it is currently confusing and we should always choose duplicate code over the wrong abstraction (problem is: it's hard to know if the abstraction is wrong when you make it 😅). 
I think we can wait changing that those.

Currently both `get_top_transactions/index.ts` and `traces/get_top_traces.ts` call `getTransactionGroups` with a `bodyQuery` param. I think it is easier to push `uiFilters` to `bodyQuery` inside `get_top_transactions/index.ts` and `traces/get_top_traces.ts` than here.